aten/src/ATen/core/class_type.cpp

```diff
@@ -524,9 +524,9 @@ void ClassType::checkNotExist(const std::string& name, const std::string& what)
 }
 
 void ClassType::addAttribute(ClassAttribute classAttribute) {
-    attributes_.push_back(classAttribute);
-    attributeTypes_.push_back(classAttribute.getType());
     AT_ASSERT(attributes_.size() == attributeTypes_.size());
+    attributeTypes_.push_back(classAttribute.getType());
+    attributes_.push_back(std::move(classAttribute));
```

Comment:
Wouldn't `emplace_back` be better here?
```suggestion
    attributeTypes_.emplace_back(classAttribute.getType());
    attributes_.emplace_back(std::move(classAttribute));
```

Reply:
@malfet probably won't make a difference here since they are the same type so there is no implicit conversion.